From 9e1b472701fa1785989f6757243a3d81e97944f2 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 14 May 2018 16:32:08 +0200
Subject: [PATCH 1/3] generate mkinitfs.conf

we may want have conditional default config depending on architecture
---
 Makefile      | 7 +++++--
 mkinitfs.conf | 2 --
 2 files changed, 5 insertions(+), 4 deletions(-)
 delete mode 100644 mkinitfs.conf

diff --git a/Makefile b/Makefile
index 68959fe..fc01d1a 100644
--- a/Makefile
+++ b/Makefile
@@ -70,12 +70,13 @@ SED_REPLACE	:= -e 's:@VERSION@:$(FULL_VERSION):g' \
 		-e 's:@sysconfdir@:$(sysconfdir):g' \
 		-e 's:@datadir@:$(datadir):g'
 
+DEFAULT_FEATURES ?= ata base cdrom ext2 ext4 keymap kms mmc raid scsi usb virtio
 
 
-all:	$(SBIN_FILES) $(SCRIPTS)
+all:	$(SBIN_FILES) $(SCRIPTS) $(CONF_FILES)
 
 clean:
-	rm -f $(SCRIPTS)
+	rm -f $(SCRIPTS) mkinitfs.conf
 
 help:
 	@echo mkinitfs $(VERSION)
@@ -116,3 +117,5 @@ install: $(SBIN_FILES) $(SHARE_FILES) $(CONF_FILES)
 		$(INSTALL) -D $$i $(DESTDIR)/usr/share/mkinitfs/$$i;\
 	done
 
+mkinitfs.conf:
+	echo 'features="$(DEFAULT_FEATURES)"' > $@
diff --git a/mkinitfs.conf b/mkinitfs.conf
deleted file mode 100644
index 67a5444..0000000
--- a/mkinitfs.conf
+++ /dev/null
@@ -1,2 +0,0 @@
-# run mkinitfs -L for a list of available features
-features="ata base cdrom ext2 ext3 ext4 keymap kms mmc raid scsi usb virtio"
-- 
2.17.0

